### PR TITLE
Removed events and replaced them with method calls

### LINKED ISSUE #156: Refactor event propagation to interface execution
The current event propagation seems to be introducing unnecessary complexity without any visible benefits. Maybe it would be worth to refactor the code to use plain old interfaces. Direct coupling would improve debuggability.

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
Fixed Travis script to start building branches

fixes #156 #211

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,15 +20,21 @@ script:
...
     MILESTONE=${MILESTONE}
     MVN_PROFILE=${MVN_PROFILE}
+    TRAVIS_BRANCH=${TRAVIS_BRANCH}
+    TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}
+    TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST}
 
     EOF
+- ./mvnw install -P docs -q -U -DskipTests=true -Dmaven.test.redirectTestOutputToFile=true
+- '[ "${MVN_GOAL}" == "deploy" ] && ./docs/src/main/asciidoc/ghpages.sh || echo "Not updating docs"'
+script:
```
Comment (adriancole) [inline .travis.yml]: nit: when changing things that seem unrelated, self-note in the PR what's going. it isn't a problem to change other things per-se, just distracting to figure out what if anything this has to do with the core change.

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/NoOpSpanReporter.java [hunk 2]
```diff
@@ -14,20 +14,18 @@
...
+ * @author Marcin Grzejszczak
  *
- * @author Spencer Gibb
+ * @since 1.0.0
  */
-@SuppressWarnings("serial")
-public class SpanContinuedEvent extends SpanContainingEvent {
+public class NoOpSpanReporter implements SpanReporter {
+	@Override
+	public void reportForCollection(Span span) {
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/NoOpSpanReporter.java]: reportForCollection is unnecessarily long and maybe lies. Not all implementations will have a collection tier, ex what if it just reports to a log file? I'd simplify by just saying report

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/SpanReporter.java [hunk 3]
```diff
@@ -14,23 +14,18 @@
...
-
-	public ClientSentEvent(Object source, Span span) {
-		super(source, span);
-	}
-
-}
\ No newline at end of file
+public interface SpanReporter {
+	/**
+	 * Reports a Span for collection
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/SpanReporter.java]: Reports a completed span out of band, usually out of process. This is typically to a trace depot (ex zipkin) or a log file.

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceFilter.java [hunk 4]
```diff
@@ -202,6 +193,19 @@ else if (skip) {
 		}
 	}
 
+	private void logServerReceived(Span parent) {
+		if (parent != null && parent.isRemote()) {
+			parent.logEvent(Span.SERVER_RECV);
+		}
+	}
+
+	private void logServerSent(SpanReporter spanReporter, Span parent) {
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceFilter.java]: I think these two methods are better inlined. For example, this method conditionally logs server sent, and also reports! The extraction doesn't help line count that much and the method name hides some pretty important details.

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java [hunk 5]
```diff
@@ -96,22 +85,16 @@ private String uriScheme(URI uri) {
...
 	/**
-	 * Close the current span and emit the ClientReceivedEvent
+	 * Close the current span and log the client received event
 	 */
 	public void finish() {
 		if (!isTracing()) {
 			return;
 		}
-		publish(new ClientReceivedEvent(this, currentSpan()));
+		currentSpan().logEvent(Span.CLIENT_RECV);
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java]: I think that this is done directly, here. I can see without navigation, that this task results in a "cr"

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/log/SpanLogger.java [hunk 6]
```diff
@@ -14,27 +14,37 @@
...
- * @author Spencer Gibb
- * @since 1.0.0
+ * @author Marcin Grzejszczak
  *
- * @see ClientSentEvent
+ * @since 1.0.0
  */
-@SuppressWarnings("serial")
-public class ServerReceivedEvent extends SpanParentContainingEvent {
+public interface SpanLogger {
```
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/log/SpanLogger.java]: it seems this component's primary concern is synchronizing MDC context with span id context. At least that's the only implementation, right?

**Comment by adriancole** [comment]
This change is certainly a step in the right direction.

SpanLogger seems the most likely to thrash (before or after this change), as the name is very generalized, yet it is designed for a very specific in-process span propagation concern. I think we should try to buff this out before 1.0

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java (hunk 5 above)
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java]: I don't get it

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/log/SpanLogger.java (hunk 6 above)
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/log/SpanLogger.java]: Yes for now MDC is the only impl other than NoOp. Maybe someone wants to add it for commons or sth (I don't know if that's even possible)

File: .travis.yml (hunk 1 above)
Comment (marcingrzejszczak) [inline .travis.yml]: Added info to the PR

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java (hunk 5 above)
Comment (adriancole) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/AbstractTraceHttpRequestInterceptor.java]: I meant to type 'I like that this is done here'! Yeah you got it
